bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java

```diff
@@ -318,19 +317,22 @@ public boolean sendTelegramPhoto(@ActionInput(name = "chatId") @Nullable Long ch
                             "Basic " + B64Code.encode(username + ":" + password, StandardCharsets.ISO_8859_1)));
                 }
                 try {
-                    ContentResponse contentResponse = request.send();
+                    // API has 10mb limit to jpg file size, without this it can only accept 2mb
+                    FutureResponseListener listener = new FutureResponseListener(request, 10 * 1024 * 1024);
+                    request.send(listener);
+                    ContentResponse contentResponse = listener.get();
                     if (contentResponse.getStatus() == 200) {
                         byte[] fileContent = contentResponse.getContent();
                         sendPhoto = new SendPhoto(chatId, fileContent);
                     } else {
                         logger.warn("Download from {} failed with status: {}", photoURL, contentResponse.getStatus());
                         return false;
                     }
-                } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                } catch (InterruptedException | ExecutionException e) {
                     logger.warn("Download from {} failed with exception: {}", photoURL, e.getMessage());
                     return false;
                 }
-            } else if (photoURL.toLowerCase().startsWith("file")) {
+            } else if (photoURL.toLowerCase().startsWith("file:") || photoURL.toLowerCase().endsWith(".jpg")) {
```

Comment:
It should be noted that `.jpeg`, `.jpe`, `.jif`, `.jfif`, and `.jfi` are all valid extensions for a file with the `image/jpeg` mimetype. So either add support for all of those or just simply check if the url contains a period (`.`) so that you can handle any extension they throw at you.

Reply:
I added support for the .jpeg extension and the latest changes are passing my tests here. As for all the extra valid extensions they will not be very common. I do see the value in supporting them and PNG, I just don't want to go there right now.

At some point my stance is that:
a) This is an improvement over what was there before and does not break backwards compatibility nor prevent future changes.
b) The file:// method already allows anything to be specified that the API supports, so there is an option that works.

As for checking if the url contains a period (.) there are file extensions that are 3 and 4 characters long so I would have to check for the period in 2 index positions of the string. Considering it is a valid use case of having raw data in base64 in this string I don't like that as a solution. Perhaps I misunderstood or another way is possible? Thank you for taking the time to help it is appreciated.

Reply:
Yeah, 10 or statements is not ideal. Typically you would put all the extensions into a list/array and then iterate through them to check if any of them is a match.
At the very least you should change the code so that you are not calling `toLowercase` 10 times.

I suggest doing something like this:
```
private static final Set<String> PHOTO_EXTENSIONS = Set.of(".jpg",".jpeg",".png",...);

private static boolean hasPhotoExt(String str){
    String lstr = str.toLowercase();
    return PHOTO_EXTENSIONS.stream().anyMatch(lstr::endsWith);
}
```

Reply:
Thanks for suggestion which I used in a tweaked way. Tested and its working at sending a png file here.